### ISSUE TITLE
Crash with strings inside ParameterizedTestParameters

### ISSUE THREAD

**Issue by athei** [Snaipe/Criterion#258]
Bug report for version: v2.3.2

I am trying to have strings (`const char *`) inside an struct that I pass to `cr_make_param_array`. When I try to access these strings I get an SIGSEGV. This is my minimal example:
```
struct str_struct {
	const char* name;
};

ParameterizedTestParameters(tester, str_test)
{
	static struct str_struct args[] = {
		{ .name = "criterion" }
	};

	return cr_make_param_array(struct str_struct, args, 1);
}


ParameterizedTest(struct str_struct *args, tester, str_test)
{
    char t = *args->name;
}
```

Output:
```
[----] /home/atheissen/Developer/secured/trusted/tests/crypto_test.c:192: Unexpected signal caught below this line!
[FAIL] tester::str_test: CRASH!
[====] Synthesis: Tested: 1 | Passing: 0 | Failing: 1 | Crashing: 1 
```

Debug:
```
Thread 1 "test_trusted" received signal SIGSEGV, Segmentation fault.
0x00005555555bbb87 in tester_str_test_impl (args=0x7ffff7fba280) at /home/atheissen/Developer/secured/trusted/tests/crypto_test.c:194
194		char t = *args->name;
(gdb) bt
#0  0x00005555555bbb87 in tester_str_test_impl (args=0x7ffff7fba280) at /home/atheissen/Developer/secured/trusted/tests/crypto_test.c:194
#1  0x00007ffff7b6aa9c in criterion_internal_test_main (fn=0x5555555bbb78 <tester_str_test_impl>) at /home/atheissen/Developer/secured/deps/Criterion/src/core/test.c:93
#2  0x00005555555bbb70 in tester_str_test_jmp () at /home/atheissen/Developer/secured/trusted/tests/crypto_test.c:192
#3  0x00007ffff7b68dd6 in run_test_child () at /home/atheissen/Developer/secured/deps/Criterion/src/core/runner_coroutine.c:227
#4  0x00007ffff7bb882e in bxfi_main () at /home/atheissen/Developer/secured/build/boxfort-build/src/boxfort/src/sandbox.c:57
#5  0x00007ffff736cb97 in __libc_start_main (main=0x7ffff7b76a1c <main>, argc=1, argv=0x7fffffffdd28, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, 
    stack_end=0x7fffffffdd18) at ../csu/libc-start.c:310
#6  0x00005555555ba10a in _start ()
(gdb) info locals
t = -35 '\335'
(gdb) info args
args = 0x7ffff7fba280
(gdb) p args->name
$11 = 0x5589cc24529e <error: Cannot access memory at address 0x5589cc24529e>
```

**Comment by Snaipe** [comment]
This is due to the side-effect that tests are run in their own child processes, and the address of `"criterion"` becomes invalid. See [this](https://github.com/Snaipe/Criterion/issues/208) issue for more information.

TL;DR: use `cr_malloc` to allocate your strings in ParameterizedTestParameters to use them in ParameterizedTest.
